### ISSUE TITLE
[Bug] Audio resampling truncates before resampling

### ISSUE THREAD

**Issue by ChiragSW** [keras-team/keras#23068]
`tf.audio.decode_wav(... desired_samples=output_sequence_length)` truncates in the source sampling rate before `tfio.audio.resample()` runs. A 48 kHz file with `sampling_rate=16000` and `output_sequence_length=16000` can produce about 5333 samples after resampling and not 16000.

repro:
```python
import os
import tempfile
import numpy as np
import tensorflow as tf
import keras

source_rate = 48000
target_rate = 16000
output_len = 16000

tmp = tempfile.mkdtemp()
wav_path = os.path.join(tmp, "sample.wav")

audio = np.random.uniform(-0.2, 0.2, size=(source_rate, 1)).astype("float32")
with open(wav_path, "wb") as f:
    f.write(tf.audio.encode_wav(audio, sample_rate=source_rate).numpy())

ds = keras.utils.audio_dataset_from_directory(
    tmp,
    labels=None,
    batch_size=None,
    shuffle=False,
    sampling_rate=target_rate,
    output_sequence_length=output_len,
)

sample = next(iter(ds))
print(sample.shape)
```

Observation:
For a 48 kHz WAV with sampling_rate=16000 and output_sequence_length=16000 keras first decodes only 16000 samples at 48 kHz which is ~0.333 sec audio. The partial audio is then resampled to 16 kHz producing roughly 5333 samples instead of 16000

Expected:
The dataset should output 16000 samples per ex after resampling assuming the input audio is long enough with shorter inputs padded with existing behavior.

**Comment by ChiragSW** [comment]
Simple fix would be when sampling_rate is provided, length normalization should happen after resampling. I will put up a PR soon
